### ISSUE TITLE
Records engine

### ISSUE THREAD

**Issue by simonhac** [opennem/opennem#214]
_tl;dr: Build a system that identifies remarkable events, and posts to a slack channel for review and further dissemination._

Records are made all the time, eg:

-  highest instantaneous renewables generation in [the NEM|NSW|VIC|QLD|SA|TAS|WA]
- rooftop solar generation exceeds 20GW for the first time
- first [dispatch interval |day|week|month] where combined renewables exceed [coal|fossils] in the NEM
- lowest gas power generation in a [week|month|year] this century
- lowest brown coal power this century 
- etc.

…but we often miss them because nobody is looking. Sometimes there's a flurry of activity to establish whether a record has been made, only to find out it's happened before or there's a false alarm.

Would be fun to build an engine with plug in record descriptions that would be smart about when to check each record, keeping track of a bunch of state.

When the engine finds a record, it should post to a slack channel where an operator can verify the event and hand on to social- or traditional media.
